garage/tf/samplers/on_policy_vectorized_sampler.py

```diff
@@ -12,16 +13,13 @@
 
 
 class OnPolicyVectorizedSampler(BatchSampler):
-    def __init__(self, algo, n_envs=None):
-        super(OnPolicyVectorizedSampler, self).__init__(algo)
+    def __init__(self, algo, n_envs=cpu_count()):
```

Comment:
i'm not sure having a default here is the best choice. defaults are good at the top-level of an API (e.g. the runner) which people actually call. If you don't expect most users to be instantiating Samplers, its better to make it a required parameter and let the code above you decide.

This is especially true for code which consumes resources (e.g. CPU) because you don't want small errors to unwittingly introduce resource-hogging bugs, which a really difficult to find.

It's much easier to debug "why am i only training on one core?" than it is "oh my god why is my machine not responding???"

That is I don't think you should have a default here. If you really need one, it should be 1.

Reply:
Make a lot of sense. Changed to 1.